public static int getInt() {
    try {
        return 1;
    } catch (Exception e) {
        return 2;
    } finally {
        return 3;
    }
}

public static void main(String[] args) {
    System.out.println(getInt());
}

Output:
3

it only returns the last value under the finally statement